src/item.h

```diff
@@ -645,6 +645,9 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
      * for other players. The player is identified by its id.
      */
     void mark_as_used_by_player(const player &p);
+    /** Marks the item as filthy, so characters with squeamish trait can't wear it. 
+    */
+    bool is_disgusted_by( const player &p ) const;
```

Comment:
I'd change it to `is_disgusting_for`, because the current name suggests that the item itself is disgusted by that filthy hobo picking it up.